sql/src/main/java/io/crate/action/sql/SQLOperations.java

```diff
@@ -72,13 +73,13 @@
     public SQLOperations(Analyzer analyzer,
                          Planner planner,
                          Provider<Executor> executorProvider,
-                         StatsTables statsTables,
+                         StatsTablesService statsTablesService,
```

Comment:
I think you could change this back to StatsTables and add a binding like this:

```
        bind(StatsTablesService.class).asEagerSingleton();
        bind(StatsTables.class).toProvider(StatsTablesService.class);
```

(and extend StatsTablesService to implement Provider<StatsTables>)

Then you can inject StatsTables directly instead of having this StatsTablesService indirection. Should make unit testing a bit easier.